packages/web-components/vite.config.ts

```diff
@@ -0,0 +1,62 @@
+import { resolve } from 'path';
+import { defineConfig } from 'vite';
+import dts from 'vite-plugin-dts';
+import { nodeResolve } from '@rollup/plugin-node-resolve';
+import vue from '@vitejs/plugin-vue';
+import pkg from './package.json' assert { type: 'json' };
+
+export default defineConfig({
+  plugins: [
+    vue({ isProduction: true, customElement: true }),
+    dts({
+      tsconfigPath: 'tsconfig.build.json'
+    }),
+    nodeResolve()
+  ],
+  build: {
+    lib: {
+      entry: resolve(__dirname, './index.ts')
+    },
+    rollupOptions: {
+      external: [
+        ...Object.keys(pkg.dependencies),
+        ...Object.keys(pkg.peerDependencies),
+        /^@prestashopcorp\/puik-components\/.*/
```

Comment:
You can remove this line too, this external dependency is already declared in your package.json